### ISSUE TITLE
Using cache when loading vocab.Vectors on Windows gives OSError Invalid Argument error

### ISSUE THREAD

**Issue by tomgun132** [pytorch/text#606]
I was trying to load fasttext common-crawl vector (.vec.gz) file into torchtext by using torchtext.vocab.Vectors
`emb = vocab.Vectors(name=fasttext_name, url=URL, cache=fasttext_cache)`
For the first time without the .pt file, loading the vectors works fine and it creates<filename> .pt file on the cache folder. However, whenever I load the vector again when the .pt file exists in the folder, it gives this kind of error:
```
 File "load_emb.py", line 94, in get_fasttext_embs
    name=fasttext_name, url=URL, cache=fasttext_cache
  File "C:\Users\user\Anaconda3\lib\site-packages\torchtext\vocab.py", line 323, in __init__
    self.cache(name, cache, url=url, max_vectors=max_vectors)
  File "C:\Users\user\Anaconda3\lib\site-packages\torchtext\vocab.py", line 432, in cache
    self.itos, self.stoi, self.vectors, self.dim = torch.load(path_pt)
  File "C:\Users\user\Anaconda3\lib\site-packages\torch\serialization.py", line 386, in load
    return _load(f, map_location, pickle_module, **pickle_load_args)
  File "C:\Users\user\Anaconda3\lib\site-packages\torch\serialization.py", line 580, in _load
    deserialized_objects[key]._set_from_file(f, offset, f_should_read_directly)
OSError: [Errno 22] Invalid argument
```

I don't have this error when I run the same code inside a linux machine. I wonder what can I do to fix this on Windows machine since I need to delete the .pt file and starts loading fasttext vector from the start again every time I need to run the code.

**Comment by zhangguanheng66** [comment]
@mttk I don't think we cover Windows now, right?

**Comment by mttk** [comment]
Windows isn't officially supported by torchtext, but I'm not convinced this should fail on Windows (as it's failing on deserialization). 
@tomgun132 What's the size of the file you're trying to load? And can you try loading the serialized file in isolation (so just a script doing `torch.load()` of the vector file), and then again like this:

```
extra_args = {"protocol":4}
torch.load(..., pickle_load_args=extra_args)
```
I'm not 100% sure of the syntax for the `extra_args`, but this should be it.

**Comment by tomgun132** [comment]
The size of the .pt file is 2.4GB. I tried loading the serialized file by just using torch.load and it fails with the same error as well. When I tried with the argument protocol, it says that `TypeError: 'protocol' is an invalid keyword argument for load()`

**Comment by mttk** [comment]
Yeah, my bad, I misremembered the `load` args, protocol is a save only arg.
This is related to Pickle having problems with loading large files (https://github.com/stanfordnlp/stanfordnlp/issues/8, https://stackoverflow.com/questions/42653386/does-pickle-randomly-fail-with-oserror-on-large-files/42725930), which exists for a while.

I've created a branch on my fork (https://github.com/mttk/text/tree/fix_vector_storage) with a hotfix. I don't have a Windows machine at hand to test it, but I tried the loading on linux and it works fine.
Could you:
1. Build torchtext from the branch 
2. Delete the stored vector file /move to another folder and download it again (the `.save` pickle protocol defaults to 2, which is bad for large files. This can't be changed retroactively.)
3. Try running the same code again, downloading the vectors and then loading from cache

If it works, it will be in the master branch soon.

**Comment by tomgun132** [comment]
I uninstalled torchtext and followed the steps you mentioned. It still doesn't work and shows the same error even with a simple `torch.load`. Though, I noticed a longer time before the code shows the error.
My python version is 3.7.3 and the model file I'm trying to load is [Japanese model of fasttext common crawl vectors](https://dl.fbaipublicfiles.com/fasttext/vectors-crawl/cc.ja.300.vec.gz) if this information can help.

**Comment by mttk** [comment]
That's odd. I'll try to get my hands on a Win machine and run some debugging in the next day or so. Thanks for the info, I'll keep you updated.

**Comment by tomgun132** [comment]
My pytorch version is 1.2.0

**Comment by mattr1** [comment]
FWIW, I ran into the same problem today. We upgraded from pytorch 0.4 to 1.2 and I get the same exception when loading a pt file (no problem when creating the pt file). The .pt file size increases by 36 bytes with torch 1.2 (2,354,897,570 bytes vs. 2,354,897,534 bytes). I also tried with @mttk's patch and still had the same exception.

I did find a fix though: It works with torch 1.1.0. Strangely, the file is still 36 bytes larger, but it works. 

And, if I create the .pt file with torch 1.2.0, I am able to load it fine with torch 1.1.0. So, the issue seems to be with something that changes in the loading, not the saving.
